src/modules/register/register.service.ts

```diff
@@ -9,14 +12,56 @@ export class RegisterService {
   constructor(
     @InjectRepository(User)
     private usersRepository: Repository<User>,
+    private readonly jwtService: JwtService,
   ) {}
 
-  async createUser(dto: CreateUserDto): Promise<User> {
-    try {
-      const user = await this.usersRepository.save(dto);
-      return user;
-    } catch (e) {
-      throw new ConflictException(e.sqlMessage);
+  async registration(dto: CreateUserDto) {
+    let candidate = await this.getUserByEmail(dto.email);
```

Comment:
try|catch everywhere where you use await

Reply:
ready